src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php

```diff
@@ -114,75 +92,67 @@ public function getFieldsAction(Request $request)
-            }
-        );
+        $view = $this->responseGetById($id, $findCallback);
 
         return $this->handleView($view);
     }
 
     /**
-     * Returns the children for a parent for the given key.
+     * Returns the sub-graph below the category which is assigned to the given parentId.
+     * This method is used by the husky datagrid to load children of a category.
+     * If request.flat is set, only the first level of the respective graph is returned in a flat format.
      *
      * @param Request $request
-     * @param mixed $key
+     * @param mixed $parentId
      *
      * @return \Symfony\Component\HttpFoundation\Response
      */
-    public function getChildrenAction(Request $request, $key)
+    public function getChildrenAction($parentId, Request $request)
     {
+        $locale = $this->getRequestParameter($request, 'locale', true);
+        // check if parent exists
+        $this->getManager()->findById($parentId, $locale);
+
         if ($request->get('flat') == 'true') {
-            $list = $this->getCategoryListRepresentation($request, $key);
+            // we need to disable the limit because husky datagrid cannot paginate nested entries
+            $list = $this->getListRepresentation($request, $locale, $parentId, true);
         } else {
-            $sortBy = $request->get('sortBy');
-            $sortOrder = $request->get('sortOrder');
-            $categoryManager = $this->get('sulu_category.category_manager');
-            $categories = $categoryManager->findChildren($key, $sortBy, $sortOrder);
-            $wrappers = $categoryManager->getApiObjects(
-                $categories,
-                $this->getRequestParameter($request, 'locale', true)
-            );
-            $list = new CollectionRepresentation($wrappers, self::$entityKey);
+            $categories = $this->getManager()->find($locale, $parentId);
+            $list = new CollectionRepresentation($categories, self::$entityKey);
         }
-        $view = $this->view($list, 200);
 
-        return $this->handleView($view);
+        return $this->handleView($this->view($list, 200));
     }
 
     /**
-     * Shows all categories
-     * Can be filtered with "parent" and "depth" parameters.
+     * Returns the whole category graph.
+     * If request.rootKey is set, only the sub-graph below the category which is assigned to the given key is returned.
+     * If request.flat is set, only the first level of the respective graph is returned in a flat format.
+     * If request.expand is set, the paths to the respective categories are expanded.
      *
      * @param Request $request
      *
      * @return \Symfony\Component\HttpFoundation\Response
      */
     public function cgetAction(Request $request)
     {
+        $locale = $this->getRequestParameter($request, 'locale', true);
+        $rootKey = $request->get('rootKey');
+        $rootId = ($rootKey) ? $this->getManager()->findByKey($rootKey, $locale)->getId() : null;
```

Comment:
That also wouldn't solve the issue from what I can see here... I think you'll need another `getListRepresentation` method, which takes a key as parameter.

Or another solution would be to not use the category key here at all. Why is this necessary? We used the id to get the children before also, didn't we?